File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.py
class Solution:
    def missingMultiple(self, nums: list[int], k: int) -> int:
        # Use a hash set for O(1) membership lookups
        num_set = set(nums)

        multiple = k

        # Increment by k until we find a value missing from the set
        while multiple in num_set:
            multiple += k

        return multiple

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var missingMultiple = function(nums, k) {
    // Add all elements to a Set for O(1) membership lookups
    const set = new Set(nums);

    let multiple = k;

    // Increment by k until we find a value missing from the set
    while (set.has(multiple)) {
        multiple += k;
    }

    return multiple;
};
